students/AleGuardia/Lesson06/mailroom_four.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+"""Takes donor and donation information and sends thank you notes. Adds donors and donations and reports it"""
+
+# Lesson 5 Mailroom Assignment by Alejandro Guardia
+
+donations = {'William Gates, III': [1000000, 585000, 5750000],
+             'Mark Zuckerberg': [15000, 5000],
+             'Jeff Bezos': [3000000],
+             'Paul Allen': [25000,1000],
+             'Elon Musk': [30000,3499]}
+
+
+main_responses = {1:"1 - Send a Thank You\n", 2:"2 - Create a Report\n",
+                  3: "3 - Send letters to everyone\n",4:"4 - quit\n"}
+main_prompt = f"Please choose one of the following:\n" \
+              f"{main_responses[1]}{main_responses[2]}{main_responses[3]}{main_responses[4]}"
+thank_you_note = "Dear {}:\n\nWe want to thank you for your generous donation of ${:.2f}.\n\n" \
+                 "It will be put to very good use.\n\n" \
+                 "\tSincerely,\n\t\tThe Team"
+
+
+def return_average(seq):
+    """Takes a sequence and returns the average"""
+    return sum(seq)/len(seq)
+
+
+def return_label():
+    """Builds and returns a table row label"""
+    label ="{:18} | {:>12} | {:>9} | {:>12}\n".format('Donor Name','Total Given','Num Gifts','Average Gift')
+    label += "-"*len(label)
+    return label
+
+
+def return_report(data):
+    """Creates and prints a report of donors and donations"""
+    report = f"{return_label()}\n"
+    for item in data:
+        row = "{:18} $ {:>12.2f}  {:>9d}  $ {:>12.2f}\n".format(item,sum(data[item]),len(data[item]),return_average(data[item]))
+        report += row
+    return report
+
+
+# Changed for loop to list Comprehension - added list comprehension
+def return_donors():
+    """Returns the names of all donors"""
+    # donors = ""
+    # for item in donations:
+    #     donors += item + "\n"
+    return [x + "\n" for x in donations]
+
+
+def prompt_donors():
+    """Prompt donor name and adds donor and donations"""
+    name = input("Please enter a donor name: ")
+    if name == 'quit':
+        return None
+    if name == 'list':
+        print(*return_donors())
+        return prompt_donors()
+    if name in donations:
+        return send_note(name,add_donation(name),thank_you_note)
+    if name not in donations:
+        add_donor(name)
+        return send_note(name,add_donation(name),thank_you_note)
+
+
+def add_donor(name):
+    """Adds a donor name"""
+    donations[name] = []
+    return donations
```

Comment:
similar comment: this function should be void based on the name it implies.

your unit test can access the global dict `donations` (`from mailroom import donations`)

note: in real life scenario this would be a bit different but accessing the module level variable `donations` works here